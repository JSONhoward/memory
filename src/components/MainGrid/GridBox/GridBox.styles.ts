import * as React from 'react';
import styled from 'styled-components'

export const StyledGridBox = React.memo(styled('div')<{col: number, row: number, match: boolean}>`
position: relative;
display: ${props => props.match ? 'none' : 'grid'};
place-items: center;
grid-column: ${props => props.col};
grid-row: ${props => props.row};
box-shadow: 1px 1px 5px rgba(0,0,0,.5);
border-radius: 5px;
background-image: linear-gradient(to top, #cfd9df 0%, #e2ebf0 100%);
cursor: pointer;
`)