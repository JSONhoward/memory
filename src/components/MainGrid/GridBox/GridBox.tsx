import React from 'react'
import { StyledGridBox } from './GridBox.styles'

interface Props {
    col: number,
    row: number,
    match: boolean
}

const GridBox: React.FC<Props> = ({col, row, match, children}) => {
    return (
        <StyledGridBox match={match} col={col} row={row}>
            {children}
        </StyledGridBox>
    )
}

export default GridBox
