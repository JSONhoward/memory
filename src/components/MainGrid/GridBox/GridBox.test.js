import React from 'react'
import {create} from 'react-test-renderer'
import GridBox from './'

const props = {
    col: 1,
    row: 1,
    match: false
}

describe('GridBox Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<GridBox />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const component = create(<GridBox {...props} />).root

        expect(component.props).toEqual(props)
    })
})