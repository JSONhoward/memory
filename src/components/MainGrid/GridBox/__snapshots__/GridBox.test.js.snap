// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridBox Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm cHysnD"
/>
`;
